src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -208,65 +177,116 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+                while psutil.pid_exists(process_id):
+                    continue
                 return
 
-            healthy_ensured_process.put(args)
+            # Put task into input_queue
+            input_queue.put(args)
             inputs, line_number, run_id = args[:3]
-            self._processing_idx[line_number] = healthy_ensured_process.format_current_process(line_number)
+
+            self._processing_idx[line_number] = format_current_process(process_name, process_id, line_number)
+            log_process_status(process_name, process_id, line_number)
 
             start_time = datetime.utcnow()
             completed = False
+            crashed = False
 
+            # Responsible for checking the output queue messages and
+            # processing them within a specified timeout period.
             while datetime.utcnow().timestamp() - start_time.timestamp() <= timeout_time:
-                try:
-                    # Responsible for checking the output queue messages and
-                    # processing them within a specified timeout period.
-                    message = healthy_ensured_process.get()
-                    if isinstance(message, LineResult):
-                        completed = True
-                        message = self._process_multimedia(message)
-                        result_list.append(message)
-                        break
-                    elif isinstance(message, FlowRunInfo):
-                        self._storage.persist_flow_run(message)
-                    elif isinstance(message, NodeRunInfo):
-                        self._storage.persist_node_run(message)
-                except queue.Empty:
-                    continue
-
-            self._completed_idx[line_number] = healthy_ensured_process.format_current_process(line_number, True)
-            # Handling the timeout of a line execution process.
-            if not completed:
-                bulk_logger.warning(f"Line {line_number} timeout after {timeout_time} seconds.")
-                ex = LineExecutionTimeoutError(line_number, timeout_time)
-                result = self._generate_line_result_for_exception(
-                    inputs, run_id, line_number, self._flow_id, start_time, ex
-                )
-                result_list.append(result)
-                self._completed_idx[line_number] = healthy_ensured_process.format_current_process(line_number, True)
+                # Monitor process aliveness.
+                crashed = self._monitor_process_alive(process_id)
+                if crashed:
+                    break
+
+                # Handle output queue message.
+                completed = self._handle_output_queue_messages(output_queue, result_list)
+                if completed:
+                    break
+
+            # Handle line execution completed.
+            if completed:
+                self._completed_idx[line_number] = format_current_process(process_name, process_id, line_number)
```

Comment:
Seems line 265, 266 are identical with line 276,277 and can be put before the if/else clause?